sim/common/sod/item_effects/phase_5.go

```diff
@@ -132,6 +133,31 @@ func init() {
 		})
 	})
 
+	// https://www.wowhead.com/classic/item=230253/hearstriker
+	// Equip: 2% chance on ranged hit to gain 1 extra attack. (Proc chance: 1%, 1s cooldown) // obviously something wrong here lol
+	core.NewItemEffect(Heartstriker, func(agent core.Agent) {
+		character := agent.GetCharacter()
+		if !character.AutoAttacks.AutoSwingRanged {
+			return
+		}
+
+		core.MakeProcTriggerAura(&character.Unit, core.ProcTrigger{
+			Name:     "Heartstrike",
+			Callback: core.CallbackOnSpellHitDealt,
+			Outcome:  core.OutcomeLanded,
+			ProcMask: core.ProcMaskRanged,
+			ICD:      time.Second * 1,
+			Handler: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+				if spell.Flags.Matches(core.SpellFlagSuppressEquipProcs) {
+					return
+				}
+				if sim.Proc(0.02, "Heartstriker") {
+					spell.Unit.AutoAttacks.ExtraRangedAttack(sim, 1, core.ActionID{SpellID: 461164})
+				}
+			},
+		})
+	})
```

Comment:
This can be simplified a bit more to:
```suggestion
		core.MakeProcTriggerAura(&character.Unit, core.ProcTrigger{
			Name:     "Heartstrike",
			Callback: core.CallbackOnSpellHitDealt,
			Outcome:  core.OutcomeLanded,
			ProcMask: core.ProcMaskRanged,
			SpellFlagsExclude: core.SpellFlagSuppressEquipProcs,
			ProcChance: 0.02,
			ICD:      time.Second * 1,
			Handler: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
				spell.Unit.AutoAttacks.ExtraRangedAttack(sim, 1, core.ActionID{SpellID: 461164})
			},
		})
	})
```